py/client/pydeephaven/_table_ops.py

```diff
@@ -62,7 +62,23 @@ def get_stub_func(cls, table_service_stub: table_pb2_grpc.TableServiceStub) -> A
         return table_service_stub.TimeTable
 
     def make_grpc_request(self, result_id, source_id) -> Any:
-        return table_pb2.TimeTableRequest(result_id=result_id, start_time_nanos=self.start_time,
+        if isinstance(self.start_time, str):
+            if isinstance(self.period, str):
+                return table_pb2.TimeTableRequest(result_id=result_id,
+                                                  start_time_string=self.start_time,
+                                                  period_string=self.period)
+
+            return table_pb2.TimeTableRequest(result_id=result_id,
+                                              start_time_string=self.start_time,
+                                              period_nanos=self.period)
+
+        if isinstance(self.period, str):
+            return table_pb2.TimeTableRequest(result_id=result_id,
+                                              start_time_nanos=self.start_time,
+                                              period_string=self.period)
+
+        return table_pb2.TimeTableRequest(result_id=result_id,
+                                          start_time_nanos=self.start_time,
```

Comment:
Probably a way we could do this w/ a single construction site:

```python
table_pb2.TimeTableRequest(
  result_id=result_id,
  start_time_nanos=self.start_time if not isinstance(self.start_time, str) else None,
  start_time_string=self.start_time if isinstance(self.start_time, str) else None,
  period_nanos=self.start_time if not isinstance(self.start_time, str) else None,
  period_string=self.start_time if isinstance(self.start_time, str) else None)
```

Would need to double check w/ the generated code to make sure `None` is treated as unset.

Reply:
This is clever, I was trying to think of how to simplify this code.